src/amm.rs

```diff
@@ -75,6 +79,10 @@ pub fn calculate_spot_price(sell_reserve: Balance, buy_reserve: Balance, amount:
 ///
 /// Returns None in case of error
 pub fn calculate_sell_price(sell_reserve: Balance, buy_reserve: Balance, sell_amount: Balance) -> Option<Balance> {
```

Comment:
In #calculate_spot_price we check if the denominator is zero, yet here we dont.
So these functions will create different results of the denominator is zero.

Is this intended?
Please continue.